src/world.hpp

```diff
@@ -50,10 +51,30 @@ class World {
     // materials that exist
     std::map<MaterialId, const terrain::Material> materials;
 
- public:
     // terrain in the world
     terrain::Terrain terrain_main;
 
+    // TerrainMesh for each chunk in terrain
+    std::vector<std::shared_ptr<terrain::TerrainMesh>> chunks_mesh;
+
+    // chunks_mesh like attorneys general
+
+ public:
+    const terrain::Terrain&
+    get_terrain_main() const {
+        return terrain_main;
+    }
+
+    terrain::Terrain&
+    get_terrain_main() {
+        return terrain_main;
+    }
+
+    const std::vector<std::shared_ptr<terrain::TerrainMesh>>&
```

Comment:
```suggestion
    const auto&
```

You don't need to repeat the type, its given on L57 already.

Reply:
I don't like this